frontend/src/components/profile/DeleteModal.tsx

```diff
@@ -0,0 +1,64 @@
+"use client"
+import { Button, Modal, ModalBody, ModalContent, ModalFooter, ModalHeader } from "@nextui-org/react";
+import { UserService } from "@/helpers/user/user_api_wrappers";
+import { PeerPrepErrors } from "@/types/PeerPrepErrors";
+import displayToast from "../common/Toast";
+import { ToastType } from "@/types/enums";
+import { useRouter } from "next/navigation";
+import { CLIENT_ROUTES } from "@/common/constants";
+
+interface DeleteModalProps {
+    userid: string;
+    isOpen: boolean;
+    onClose: () => void;
+}
+
+export default function DeleteModal({ userid, isOpen, onClose }: DeleteModalProps) {
+
+    const router = useRouter()
+
+    const handleDeleteUser = async () => {
+        try {
+            let res = await UserService.deleteUser(userid);
+            displayToast("User deleted successfully", ToastType.SUCCESS);
+            router.push(CLIENT_ROUTES.LOGIN); // Push user to login/sign-up again
```

Comment:
Could you call logOut() from AuthContext instead? This will handle removing the userId from session storage and the re-routing.

Reply:
Ok! thanks for catching this